api/users.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const router = express.Router();
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt')
+const { JWT_SECRET } = process.env;
+const { createUser, getUser, getUserByUsername } = require('../db');
+const { requireUser } = require('./utils');
+
+
+// users/
+router.get('/', async (req, res, next) => {
+    const users = await getUser();
+    try {
+        res.send({
+            users
+        })
+    } catch (error) {
+        next(error)
+    }
+});
+
+// users/my account
+router.get('/myaccount', requireUser, async (req, res, next) => {
+            res.send({
+                ...req.user
+            })
+})
+
+// users/register
+router.post('/register', async (req, res, next) => {
+    const { username, password, name, email } = req.body
+    try {
+        const _user = await getUserByUsername(username)
+        if (_user) {
+            next({
+                name: 'UserExistsError',
+                message: 'A user by that username already exists'
+            })
```

Comment:
This looks good 👍 
Maybe we should also write a getUserByEmail function that we can include in this if - elseif structure to tell the user that an account with that email already exists as well